collector/set.go

```diff
@@ -84,11 +98,24 @@ func (s *Set) Collect(ch chan<- prometheus.Metric) {
 func (s *Set) Describe(ch chan<- *prometheus.Desc) {
```

Comment:
Refactoring `Describe` as well is probably overkill (i mean, sending a var down a channel isn't exactly a heavy operation), but having only `Collect` use an errgroup felt weird :D